### PR TITLE
[Math] added MDCCeilScaled and MDCFloorScaled (#1917)

### LINKED ISSUE #1917: [Math] Implement MDCCeilScaled()
Throughout Text Fields there are many instances of the following:

CGFloat scale = UIScreen.mainScreen.scale;
CGFloat placeholderEstimatedHeight = 
    MDCCeil(self.textInput.placeholderLabel.font.lineHeight * scale) / scale;

We should consider if we want to pull this into MDCMath.

Name and interface TBD.

### PR DESCRIPTION
**Pull request by fluxminds**
closes https://github.com/material-components/material-components-ios/issues/1917

### CODE REVIEW

File: components/private/Math/tests/unit/MDCMathTests.m [hunk 1]
```diff
@@ -23,6 +23,32 @@ @interface MDCMathTests : XCTestCase
 
 @implementation MDCMathTests
 
+#pragma mark - MDCCeilScaled and MDCFloorScaled
+
+- (void)testMDCFloorScaled {
```
Comment (romoore) [inline components/private/Math/tests/unit/MDCMathTests.m:28]: Please add more tests and structure them in the Given-When-Then style that we try to encourage in general. We prefer to have 1 test for each test condition whenever possible.

File: components/private/Math/tests/unit/MDCMathTests.m [hunk 2]
```diff
@@ -23,6 +23,32 @@ @interface MDCMathTests : XCTestCase
...
+  CGFloat inputNumber = 1.3f;
+
+  CGFloat scale1 = 2.0f;
+  CGFloat expectedOutputNumber1 = 1.0f;
+  XCTAssertEqual(MDCFloorScaled(inputNumber, scale1), expectedOutputNumber1);
```
Comment (romoore) [inline components/private/Math/tests/unit/MDCMathTests.m]: Please use XCTAssertEqualWithAccuracy(...) when comparing floating point values.

File: components/private/Math/src/MDCMath.h [hunk 3]
```diff
@@ -131,6 +131,22 @@ static inline CGFloat MDCSqrt(CGFloat value) {
 #endif
 }
 
+static inline CGFloat MDCCeilScaled(CGFloat value, CGFloat scale) {
```
Comment (romoore) [inline components/private/Math/src/MDCMath.h]: Please add documentation here and in `MDCFloorScaled` to describe how these work, particularly with different values of `scale`.

**Review by romoore — APPROVED** [review]

**Comment by fluxminds** [comment]
@romoore Thanks for the review. Comments have been addressed in 911f1233ba1bddafad0b5f4388bb414b3bcf2347

File: components/private/Math/tests/unit/MDCMathTests.m (hunk 1 above)
Comment (romoore) [reply components/private/Math/tests/unit/MDCMathTests.m:28]: Please add tests for negative values as well.

Overall looks good!

**Review by romoore — APPROVED** [review]
